Engine/Settings.cs

```diff
@@ -296,167 +252,309 @@ private bool IsStringOrStringArray(object val)
-                                    Strings.WrongValueHashTable,
-                                    val,
-                                    key));
-                        }
-                    }
+                    exceptions.Add(new InvalidDataException(string.Format(
+                        Strings.SettingValueElementIsNull,
+                        settingName,
+                        currentElementIndex)));
+                    continue;
                 }
-                else
+
+                if (!(element is string))
                 {
-                    throw new InvalidDataException(
-                            string.Format(
-                                CultureInfo.CurrentCulture,
-                                Strings.WrongValueHashTable,
-                                val,
-                                key));
+                    exceptions.Add(new InvalidDataException(string.Format(
+                        Strings.SettingValueElementIsNotStringType,
+                        settingName,
+                        currentElementIndex,
+                        element)));
+                    continue;
                 }
+
+                strings.Add(element as string);
             }
 
-            return values;
+            return strings;
         }
 
-        /// <summary>
-        /// Sets the arguments for consumption by rules
-        /// </summary>
-        /// <param name="ruleArgs">A hashtable with rule names as keys</param>
-        private Dictionary<string, Dictionary<string, object>> ConvertToRuleArgumentType(object ruleArguments)
+        private bool? ParseSettingValueBoolean(object value, string settingName, IList<Exception> exceptions)
         {
-            var ruleArgs = ruleArguments as Dictionary<string, object>;
-            if (ruleArgs == null)
+            if (value == null)
             {
-                throw new ArgumentException(Strings.SettingsInputShouldBeDictionary, nameof(ruleArguments));
+                exceptions.Add(new InvalidDataException(string.Format(
+                    Strings.SettingValueIsNull,
+                    settingName)));
+                return null;
             }
-
-            if (ruleArgs.Comparer != StringComparer.OrdinalIgnoreCase)
+            
+            if (!(value is bool))
             {
-                throw new ArgumentException(Strings.SettingsDictionaryShouldBeCaseInsesitive, nameof(ruleArguments));
+                exceptions.Add(new InvalidDataException(string.Format(
+                    Strings.SettingValueIsNotBooleanType,
+                    settingName,
+                    value)));
+                return null;
             }
 
-            var ruleArgsDict = new Dictionary<string, Dictionary<string, object>>(StringComparer.OrdinalIgnoreCase);
-            foreach (var rule in ruleArgs.Keys)
+            return (bool) value;
+        }
+
+        private void ParseSettingsHashtable(Hashtable settings)
```

Comment:
Should this really be made `public` without also making it `static`? If so, then I will go ahead and make the change. If not, then if this should be changed to be both `public` and `static`, then I would ask to postpone this change to a subsequent pull request in which additional related refactorings can be made. This is because--as far as I understand it--rendering a `Settings` object constructible via a static `Parse...(Hashtable)` method (i.e. a factory method) should be coupled with deprecating its constructibility from a yet-to-be-parsed `Hashtable` via a public constructor, and thus also refactoring any extant client logic (e.g. in the "Engine/Settings/InvokeScriptAnalyzerCommands.cs" file).